release-notes-1.2.0.md

```diff
@@ -0,0 +1,761 @@
+
+```ballerina
+import ballerina/lang.'boolean;
+
+public function main() {
+    boolean|error b = 'boolean:fromString("true");
+}
+```
+
+#### Enhanced `lang.int`, `lang.string`, `lang.array`, and `lang.map` modules
+
+The above-mentioned lang library modules have been enhanced by adding new API elements (e.g., constants and functions). 
+
+- `lang.int` module - Constants added: for max/min values of the `int` type 
+- `lang.string` module - New functions added: `lastIndexOf()`, `equalsIgnoreCaseAscii()`, `toCodePointInt()`, `fromCodePointInt()`
+- `lang.array` module - New functions added: `lastIndexOf()`, `toStream()`
+- `lang.map` module - New functions added: `toArray()`, `removeIfHasKey()`
+
+## Runtime
+
+### Compilation time improvement
+
+Major improvements to the compilation (code-generation) time are implemented. The compiler backend, which generates the Java bytecode for Ballerina programs is now rewritten to improve the overall compilation time. This change has specifically reduced the total time it takes for Java bytecode generation.
+
+### The `@strand` annotation
+
+As per the Ballerina language specification, new strands should belong to the same thread as the currently executing strand. Strands, which belong to the same thread share the following characteristics. 
+
+- Only one strand can run simultaneously.
+- The currently-running strand should yield to switch its thread to execute another strand.
+- Strands belonging to a particular thread are cooperatively multitasked.
+- All the strands created in a particular execution of a Ballerina program will be executed sequentially by a single thread. 
+
+In some situations, it would be desirable to assign strands to different threads. You can change this behaviour with an annotation but such an annotation was not available in Ballerina releases prior to 1.2.0. This has been a major spec deviation. 
+
+A major revamp of our strand scheduler was done in this release to address the spec deviation. 
+
+## Standard Library
+
+### Enhanced HTTP module
+
+#### Cookie support
+
+- Enable the server to read cookies from the request and write cookies to the response (in "Set-Cookie" header).
+- Enable configuring the client to enable/disable cookies, enable/disable persistent cookies, and whether to block/unblock third-party cookies.
+- Add persistent cookies support and add support to plug custom persistent storages.  
+
+#### Trailer support
+
+Add support to create, read, update, and delete trailing headers in the response message. All existing header functions are changed to accept a finite-typed variable. The default value is set as `leading`, which means by default, those functions apply to leading headers. In order to apply trailer headers, you need to set the position as trailing as shown below.
+
+```ballerina
+string trailerHeader = response.getHeader("foo", position = "trailing");
+```
+#### Cache support
+
+HTTP Caching by default is disabled.
```

Comment:
Not sure this is the correct place list this. This is a change from the current behaviour. Not a feature addition/improvement

Reply:
@daneshk - WDYT?

Reply:
Yes, This is a change in current behavior. having a new section doesn't make sense. Shall we rename `Cache support` to `Other changes` and list it this under that

Reply:
#### Other changes
- HTTP Caching by default is disabled.

Reply:
@daneshk - Can you add the changes here itself so I can commit directly?

Reply:
@daneshk - Can you add your change as a direct comment that I can commit?

Reply:
done

Reply:
Maybe add a subheading under the HTTP module called "Changes in behavior" and mention that in previous versions HTTP caching was enabled by default, but in 1.2.0, it's disabled by default

Reply:
@pubudu91 - Please send further changes or suggestions in another PR as this is already merged.